src/main/java/com/amazon/ion/impl/macro/EExpressionArgsReader.java

```diff
@@ -231,15 +231,16 @@ private void readStreamAsExpressionGroup(
      * @param expressions receives the expressions as they are materialized.
      */
     protected void readValueAsExpression(boolean isImplicitRest, List<Expression.EExpressionBodyExpression> expressions) {
-        if (isMacroInvocation()) {
+        if (isImplicitRest && !isContainerAnExpressionGroup()) {
+            readStreamAsExpressionGroup(expressions);
+            return;
+        } else if (isMacroInvocation()) {
             collectEExpressionArgs(expressions); // TODO avoid recursion
             return;
         }
         IonType type = reader.encodingType();
         List<SymbolToken> annotations = getAnnotations();
-        if (isImplicitRest && !isContainerAnExpressionGroup()) {
-            readStreamAsExpressionGroup(expressions);
-        } else if (IonType.isContainer(type)) {
+        if (IonType.isContainer(type) && !reader.isNullValue()) {
```

Comment:
🗺️ Added a check for null here because in the `Expression` model, all null values are treated as scalars. This was causing some of the conformance tests to fail for e.g. `make_list`.